glom/core.py

```diff
@@ -753,6 +753,90 @@ def __repr__(self):
         return '%s(%r, args=%r, kwargs=%r)' % (cn, self.func, self.args, self.kwargs)
 
 
+class Partial(object):
+    """
+    makes "upgrading" arbitrary python functions into
+    glom specs more convenient, so that itertools, toolz, et al
+    may be more easily integrated into streaming glom chains
+    """
+    def __init__(self, func, specfunc=None):
+        self.func, self.subspec = func, specfunc
+        self.args = ()
+
+    @classmethod
+    def specfunc(cls, specfunc):
+        return cls(None, specfunc)
+
+    def consts(self, *a, **kw):
+        """pass *a and **kw to func"""
+        ret = Partial(self.func, self.subspec)
+        ret.args = self.args + ('C', a, kw)
+        return ret
+
+    def specs(self, *a, **kw):
+        """glom all of *a and **kw and pass to func"""
+        ret = Partial(self.func, self.subspec)
+        ret.args = self.args + ('S', a, kw)
+        return ret
+
+    def star(self, args=None, kwargs=None):
+        """
+        glom args and kwargs and pass to func as *a and **kw
+        """
+        assert args is not None or kwargs is not None
```

Comment:
I'm probably gonna go in and turn these asserts into TypeErrors.

Reply:
sounds good... maybe a non-standard `ArgumentError`?

Reply:
Usually I stick to TypeErrors for spec construction like this. I switch to GlomError derivatives for execution time.